### PR TITLE
Github Actions

### PR DESCRIPTION
**Pull request by parsonsmatt**
Before submitting your PR, check that you've:

- [ ] Bumped the version number
- [ ] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [ ] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddock

After submitting your PR:

- [ ] Update the Changelog.md file with a link to your PR
- [ ] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

File: .github/workflows/haskell.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-haskell@v1.1.2
+        id: setup-haskell-cabal
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: ${{ matrix.cabal }}
+      - uses: harmon758/postgresql-action@v1
+        with:
+           postgresql version: '12'  # See https://hub.docker.com/_/postgres for available versions  
+           postgresql user: persistent
```
Comment (parsonsmatt) [inline .github/workflows/haskell.yml:29]:
```suggestion
           postgresql user: esqutest
           postgresql password: esqutest
```

File: .github/workflows/haskell.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+           postgrseql db: persistent
+      - uses: mirromutth/mysql-action@v1.1
+        with:
+          mysql version: '8.0' # Optional, default value is "latest". The version of the MySQL
+          mysql database: 'persistentt' # Optional, default value is "test". The specified database which will be create
```
Comment (parsonsmatt) [inline .github/workflows/haskell.yml:34]:
```suggestion
          mysql database: 'travis' # Optional, default value is "test". The specified database which will be create
```

**Comment by belevy** [comment]
so if you want to use stack `stack build --only-dependencies` would probably be helpful for caching

File: .github/workflows/haskell.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        ghc: ["8.6.5", "8.8.3", "8.10.1"]
+    env:
+      CONFIG: "--enable-tests --enable-benchmarks"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-haskell@v1.1.2
+        id: setup-haskell-cabal
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          cabal-version: ${{ matrix.cabal }}
```
Comment (belevy) [inline .github/workflows/haskell.yml:25]:
```suggestion
          ghc-version: ${{ matrix.ghc }}
          stack-version: ${{ matrix.stack }}
          enable-stack: true
          stack-no-global: true
          stack-setup-ghc: true
```

File: .github/workflows/haskell.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  pull_request:
+    types:
+      - opened
+      - synchronize
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        cabal: ["3.2"]
```
Comment (belevy) [inline .github/workflows/haskell.yml:15]:
```suggestion
        stack: ["latest"]
```

File: .github/workflows/haskell.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          path: |
+            ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+            dist-newstyle
+          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.ghc }}-
+      - run: cabal v2-build $CONFIG
+      - run: cabal v2-test $CONFIG
+      - run: cabal v2-haddock $CONFIG
+      - run: cabal v2-sdist
```
Comment (belevy) [inline .github/workflows/haskell.yml:51]: not sure what these should be

**Pull request merged**